### DISCUSSION TITLE
Is it no longer possible to staticly assign an IP to a new VM?

### DISCUSSION THREAD

**Discussion by mhamzahkhan** [oVirt/terraform-provider-ovirt#469] [Q&A] [answered]
Previously, in the v0 version of the provider, we had the initialization and nic_configuration options to assign an IP statically to a VM which made oVirt generate the relevant bits for cloud-init to set the IP of the VM statically.

Looking through the docs, now the only initialization options are initialization_hostname and initialization_custom_script, which would require a custom cloud-init script. 

Is there no way to continue to use the ovirt options to assign a static IP?

**Comment by ghost** [comment, marked as answer]
Hello @mhamzahkhan , you are correct, there is currently no way to assign static IP addresses through initialization.

As we described in the [announcement](https://blogs.ovirt.org/2021/10/important-changes-to-the-ovirt-terraform-provider/), the new provider **does not have the full functionality** of the old one. This is a tradeoff between quality and functionality. The new provider has rigorous tests and we have fixed dozens of bugs that have cropped up, but we did not have the resources to implement every feature that there was, although untested, in the old provider. **You can still use the old provider**, and we can send pull requests to it and we will merge them if they look ok, but **we won't be developing it any further ourselves**.

---

If you are interested in implementing assigning static IP addresses, you can contribute to the new provider. You need to implement it in the following places:

1. In [go-ovirt-client](https://github.com/oVirt/go-ovirt-client) take a look at the [VM creation](https://github.com/oVirt/go-ovirt-client/blob/main/vm_create.go#L61). Here you need to add the fields to add a static IP. Don't forget, we need tests and a mock functionality for it to merge the change. Read the [contribution guide](https://github.com/oVirt/go-ovirt-client/blob/main/CONTRIBUTING.md) for details on how to do this.
2. Once go-ovirt-client has the functionality in, you can implement the change in this repository. This change would likely need to be implemented [here](https://github.com/oVirt/terraform-provider-ovirt/blob/main/internal/ovirt/resource_ovirt_vm.go). As before, we have a [detailed contribution guide](https://github.com/oVirt/terraform-provider-ovirt/blob/main/CONTRIBUTING.md) for implementing this change.

Thanks to the tests, we can now merge changes fairly quickly. If the changes are ok and have tests, we can usually release a new version of the provider within 1-2 weeks of receiving the patches maximum, but we had changes roll out within 1-2 days too.

I'm sorry I can't give you the answer you are probably looking for, but I hope this still helps.
